docs/extend/admin.md

```diff
@@ -1 +1,159 @@
+
+Before you can register anything, you need to tell `ExtensionData` what extension it is about to get data for. 
+
+Simply run the `for` function on `app.extensionData` passing in the id of your extension. To find you extension id, take the composer name and replace any slashes with dashes (example: 'fof/merge-discussions' becomes 'fof-merge-discussions').
+
+For the following example, we will use the fictitious extension 'acme/interstellar':
+
+```js
+
+app.initializers.add('interstellar', function(app) {
+
+  app.extensionData
+    .for('acme-interstellar')
+});
+```
+
+Once that is done, you can begin adding settings and permissions. 
+
+::: Note
+All registration functions on `ExtensionData` are chainable, meaning you can call them one after another without running `for` again.
+:::
+
+### Registering Settings
+
+Adding settings fields in this way is recommended for simple items. As a rule of thumb, if you only need to store things in the settings table, this should be enough for you.
+
+To add a field, call the `registerSetting` function after `for` on `app.extensionData` and pass a 'setting object' as the first argument. Behind the scenes `ExtensionData` actually turns your settings into an [`ItemList`](https://api.docs.flarum.org/js/master/class/src/common/utils/itemlist.ts~itemlist), you can pass a priority number as the second argument. 
+
+Here's an example with a switch (boolean) item:
+
+```js
+
+app.initializers.add('interstellar', function(app) {
+
+  app.extensionData
+    .for('acme-interstellar')
+    .registerSetting(
+      {
+        setting: 'acme-interstellar.coordinates', // This is the key the settings will be saved under in the settings table in the database.
+        label: app.translator.trans('acme-interstellar.admin.coordinates_label'), // The label to be shown letting the admin know what the setting does.
+        type: 'boolean', // What type of setting this is, valid options are: boolean, text (or any other <input> tag type), and select. 
+      },
+      30 // Optional: Priority
+    )
+});
+```
+
+If you use `type: 'select'` the setting object looks a little bit different:
+
+```js
+{
+  setting: 'acme-interstellar.fuel_type',
+  label: app.translator.trans('acme-interstellar.admin.fuel_type_label'),
+  type: 'select',
+  options: {
+    'LOH': 'Liquid Fuel', // The key in this array is what the setting will be stored as in the database, the value is the label the admin will see (remember to use translations if they make sense in your context).
+    'RDX': 'Solid Fuel',
+  },
+  default: 'LOH',
+}
+```
+
+
+If you want to add something to the settings like some extra text or a more complicated input, you can also pass a callback as the first argument that returns JSX. This callback will be executed in the context of [`ExtensionPage`](https://api.docs.flarum.org/js/master/class/src/admin/components/extensionpage.js~extensionpage) and setting values will not be automatically serialized.
```

Comment:
Could you include an example, preferably one that binds to the setting val? Just a simple input would be fine